probeinfo_api.yaml

```diff
@@ -294,6 +294,10 @@ components:
             $ref: "#/components/schemas/RetentionDays"
           encryption:
             $ref: "#/components/schemas/Encryption"
+          moz_pipeline_metadata_defaults:
+            $ref: "#/components/schemas/MozPipelineMetadata"
+          moz_pipeline_metadata:
+            $ref: "#/components/schemas/MozPipelineMetadataOverride"
```

Comment:
I notice that you chose the name `MozPipelineMetadataOverride` for this. I do worry a little bit about the naming of `moz_pipeline_metadata` and `moz_pipeline_metadata_defaults`. Do you feel like `moz_pipeline_metadata_overrides` would be a more helpful/explicit name?

Reply:
I followed naming that was suggested in the proposal but I do think that `moz_pipeline_metadata/moz_pipeline_metadata_overrides` is a better structure (the more specific naming is lower in the tree) , except when I consider having no `moz_pipeline_metadata` at the top level and only have ping specific pipeline_metadata configured, then the naming `moz_pipeline_metadata_overrides` doesn't make sense.  I'll explore some options and update here.

Reply:
Another option would be to mash the two together like:

```
moz_pipeline_metadata:
  all-pings: # here are the defaults
  specific-ping-name: # here are overrides for specific-ping-name
  other-ping: ...
```

[`all-pings` is a reserved name in Glean internals](https://mozilla.github.io/glean/dev/core/internal/reserved-ping-names.html?highlight=all-pings#reserved-ping-names) so no application can ever define a ping with that name. So we'd be following that same format. Then we'd have the `probe_scraper` code handle `all-pings` specially as the place where defaults are defined. This feels a little bit magical, though.

Reply:
cc @Dexterp37 for gut reaction to the idea of using `all-pings` as a magic key for defining defaults here.

Reply:
Also, we don't have to block on this naming discussion. I'm fine with merging this as it exists now so you can confidently work on the next pieces. It's still possible to circle back and modify these names fairly easily up to the point where we start relying on the interface in mozilla-schema-generator.

Reply:
What if we don't include `all-pings`?  The documentation would state that moz_pipeline_metadata sets metadata for all pings and describe how to set for specific pings (override or single config).

```
moz_pipeline_metadata:
  expiration_policy:
    delete_after_days: 18
  specific-ping-name: # here are overrides for specific-ping-name
    expiration_policy:
      delete_after_days: 30
  other-ping: ...
```
This case looks readable to me as well:
```
moz_pipeline_metadata:
    specific-ping-name: # here are overrides for specific-ping-name
    expiration_policy:
      delete_after_days: 30
```

Reply:
I think it would be difficult/confusing to mix the ping-specific sections at the same level (in JSON schema terms, we would have both properties and additionalProperties) - particularly imagine that an app defined a ping named `expiration_policy` which would be ambiguous. But I think providing a specific field for per-ping overrides nested under `moz_pipeline_metadata` would be reasonable. So that would be like:

```
moz_pipeline_metadata:
  expiration_policy:
    delete_after_days: 18
  overrides:
    my-specific-ping:
      expiration_policy:
        delete_after_days: 30
```

Again the `overrides` name is up for debate. Could be `ping_specific`, `by_ping`, `per_ping`, `pings`, ... ?

Reply:
We can determine if any changes are needed.  In the meantime I did update the definition name and ref definitions to match (moz_pipeline_metadata_defaults -> MozPipelineMetadataDefaults, moz_pipeline_metadata -> MozPipelineMetadata)

Reply:
> cc @Dexterp37 for gut reaction to the idea of using `all-pings` as a magic key for defining defaults here.

While it's true that it's reserved, I'd stay away from this name to minimize confusion.  Maybe something like `default-for-all-pings`? Or `all-pings-defaults`? or `base`?

Reply:
Current plan (or nested field for overrides) look both reasonable to me !